### PR TITLE
Websocket +jms + logging

### PR DESCRIPTION
**Pull request by mimaboci**
unfinished

### CODE REVIEW

**Review by mkralik3 — APPROVED** [review]

File: persistence/src/main/java/cz/fi/muni/pv243/entity/ParserConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+import javax.persistence.Table;
+
+/**
+ * 
+ * @author Michaela Bocanova
+ *
+ */
+@Entity
+@Table(name = "PARSERCONFIGURATION")
+public class ParserConfiguration implements Serializable {
```
Comment (ghost) [inline persistence/src/main/java/cz/fi/muni/pv243/entity/ParserConfiguration.java]: Not sure why we need this Class. It contains basically the same properties as Parser class. "confirmed" can be added

File: persistence/src/main/java/cz/fi/muni/pv243/infinispan/store/CachedParserConfigurationStore.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    @Transactional
+    public ParserConfiguration addParser(ParserConfiguration parser){
+        parser = delegate.addParser(parser);
+        parserCache.put(parser.getId(), parser);
+        return parser;
+    }
+
+    @Override
+    public ParserConfiguration findParser(Long id) {
+        return delegate.findParser(id);
```
Comment (ghost) [inline persistence/src/main/java/cz/fi/muni/pv243/infinispan/store/CachedParserConfigurationStore.java]: Should look for id in cache first and then use JPA

File: persistence/src/main/java/cz/fi/muni/pv243/infinispan/store/CachedParserStore.java [hunk 3]
```diff
@@ -38,4 +38,9 @@ public Parser addParser(Parser parser){
         parserCache.put(parser.getId(), parser);
         return parser;
     }
+
+    @Override
+    public Parser findParser(Long id) {
+        return delegate.findParser(id);
```
Comment (ghost) [inline persistence/src/main/java/cz/fi/muni/pv243/infinispan/store/CachedParserStore.java]: The same as for ParserConfiguration.

File: pom.xml [hunk 4]
```diff
@@ -11,7 +9,9 @@
     <modules>
         <module>persistence</module>
         <module>rest</module>
-    </modules>
+        <module>service</module>
+    <module>ws</module>
```
Comment (ghost) [inline pom.xml]: wrong indentation

File: service/src/main/java/cz/fi/muni/pv243/service/ParserConfigurationServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    @Inject
+    @CachedStore
+    private CachedParserConfigurationStore configStore;
+    
+    @Override
+    public void confirm(ParserConfiguration config) {
+    	Parser parser = parserStore.findParser(config.getRestaurantId());
+    	if (parser != null) {    		
+    		parser.setXpath(config.getXpath());
+    		//parserStore.addParser(parser);//update
```
Comment (ghost) [inline service/src/main/java/cz/fi/muni/pv243/service/ParserConfigurationServiceImpl.java]: Why is this commented and not used updateParser()?

File: service/src/main/java/cz/fi/muni/pv243/service/ParserConfigurationServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+    	}
+    	else {
+    		parser = new Parser();
+    		//settery
```
Comment (ghost) [inline service/src/main/java/cz/fi/muni/pv243/service/ParserConfigurationServiceImpl.java]: Not adding xpath to newly created parser

**Review by ghost — COMMENTED** [review]
Hi, I was not able to deploy it to wildfly, could you please add some README to ws module? Also, I find this ws implementation little bit messy (lots of code which is not used for now), which is quite understandable since we don't know exactly how and what for it will be used. That's also the reason why I think we shouldn't merge this for now. Instead of this, we should wait for Javascript and then make it working final solution, based on this PR. WDYT @mimaboci?

File: persistence/src/main/java/cz/fi/muni/pv243/entity/ParserConfiguration.java (hunk 1 above)
Comment (mimaboci) [reply persistence/src/main/java/cz/fi/muni/pv243/entity/ParserConfiguration.java]: ok, I deleted the entity ParserConfiguration

File: persistence/src/main/java/cz/fi/muni/pv243/infinispan/store/CachedParserStore.java (hunk 3 above)
Comment (mimaboci) [reply persistence/src/main/java/cz/fi/muni/pv243/infinispan/store/CachedParserStore.java]: like this?
Parser parser = parserCache.get(id);
return parser == null ? delegate.findParser(id) : parser;

File: pom.xml (hunk 4 above)
Comment (mimaboci) [reply pom.xml]: don't understand what happened here. the indentation is correct on my side.

File: persistence/src/main/java/cz/fi/muni/pv243/infinispan/store/CachedParserConfigurationStore.java (hunk 2 above)
Comment (mimaboci) [reply persistence/src/main/java/cz/fi/muni/pv243/infinispan/store/CachedParserConfigurationStore.java]: deleted

File: persistence/src/main/java/cz/fi/muni/pv243/infinispan/store/CachedParserStore.java (hunk 3 above)
Comment (ghost) [reply persistence/src/main/java/cz/fi/muni/pv243/infinispan/store/CachedParserStore.java]: that is exactly what we need, but with Java 8 we can use:

````java
    @Override
    public Restaurant findById(String googleID) {
        return parserCache.computeIfAbsent(googleID,
                s -> delegate.findById(s));
    }
````

File: persistence/src/main/java/cz/fi/muni/pv243/entity/ParserConfiguration.java (hunk 1 above)
Comment (mimaboci) [reply persistence/src/main/java/cz/fi/muni/pv243/entity/ParserConfiguration.java]: but foreign key in parser then looks like this:
@ManyToOne(fetch = FetchType.EAGER, cascade = CascadeType.ALL)
@JoinColumn(name="RESTAURANT_ID")
Restaurant restaurant;

File: pom.xml (hunk 4 above)
Comment (ghost) [reply pom.xml]: Check tabs and spaces

File: persistence/src/main/java/cz/fi/muni/pv243/entity/ParserConfiguration.java (hunk 1 above)
Comment (ghost) [reply persistence/src/main/java/cz/fi/muni/pv243/entity/ParserConfiguration.java]: And it is problem? We will then just need to make sure that there is only one parser for restaurant and day with confirmed status. What are advantages of having two entities? I think we will need to have ManyToOne relationship anyway, as we need to have parser for each day.

File: persistence/src/main/java/cz/fi/muni/pv243/infinispan/store/CachedParserStore.java (hunk 3 above)
Comment (mimaboci) [reply persistence/src/main/java/cz/fi/muni/pv243/infinispan/store/CachedParserStore.java]: fixed

File: persistence/src/main/java/cz/fi/muni/pv243/entity/ParserConfiguration.java (hunk 1 above)
Comment (mimaboci) [reply persistence/src/main/java/cz/fi/muni/pv243/entity/ParserConfiguration.java]: there's unique constraint on column confirmed in the latest commit

**Comment by mkralik3** [comment]
resolved https://github.com/mkralik3/pv243-ParserManager/pull/7

**Pull request closed without merging**